exe/kubernetes-render

```diff
@@ -13,21 +13,25 @@ ARGV.options do |opts|
   opts.on("--bindings=BINDINGS", "Expose additional variables to ERB templates " \
     "(format: k1=v1,k2=v2, JSON string or file (JSON or YAML) path prefixed by '@')") { |b| parser.add(b) }
   opts.on("--template-dir=DIR", "Set the template dir (default: config/deploy/$ENVIRONMENT).") do |d|
-    template_dir = d
+    template_dir = [d]
   end
   opts.parse!
   bindings = parser.parse
 end
 
 templates = ARGV
 logger = KubernetesDeploy::FormattedLogger.build(verbose_prefix: false)
+if template_dir && !File.directory?(template_dir.first)
```

Comment:
You're right, this doesn't handle the default case of using `config/deploy/$ENVIRONMENT`.
Unintuitively, the template directory is passed to `RenderTask#new` whereas the list of filenames is passed as an (optional) argument to `RenderTask#run`. This doesn't square with how we want `-f` to be used so IMO precludes implementing that feature until we make the move over to Krane, where we have the opportunity to make such a breaking change without having to accommodate this older method.

Reply:
Yes, that's what it was trying to do, but failed since it wouldn't transparently use `config/deploy/$ENVIRONMENT` if no template dir argument was given (it would erroneously fail). There's a cleaner way to do this in the end, which is setting `template_dir = []` instead of nil and proceeding as normal. `with_validated_template_paths` should do the right thing in that case and maintain compatibility with the CLI/API interface of the render task. Note that render task still only accepts a _single_ template directory, since we can't change that without breaking the current API (specifically, the trailing optional list of filenames)